src/nidmm/system_tests/test_system_nidmm.py

```diff
@@ -244,3 +244,64 @@ def test_configure_waveform_acquisition(session):
     assert session.range == 100
     assert session.waveform_rate == 100000
     assert session.waveform_points == 400
+
+
+def test_fetch_waveform(session):
+    session.configure_waveform_acquisition(nidmm.Function.WAVEFORM_VOLTAGE, 10, 1000, 10)
+    with session.initiate():
+        number_of_points_to_read = 10
+        measurements,actual_number_of_points = session.fetch_waveform(number_of_points_to_read)
+        assert len(measurements) == number_of_points_to_read
+        assert actual_number_of_points == number_of_points_to_read
+
+
+def test_fetch_waveform_error(session):
+    try:
+        session.configure_waveform_acquisition(nidmm.Function.WAVEFORM_VOLTAGE, 10, 1000, 10)
+        with session.initiate():
+            number_of_points_to_read = 100
+            session.fetch_waveform(number_of_points_to_read)   # trying to fetch points more than configured
+    except nidmm.Error as e:
+        assert e.code == -1074126845  # Max Time exceeded before operation completed
+
+
+def test_get_measurement_period():
+        with nidmm.Session('FakeDevice', False, True, 'Simulate=1, DriverSetup=Model:4072; BoardType:PXI') as session:
+            session.configure_measurement_digits(nidmm.Function.DC_VOLTS, 10, 5.5)
+            measurement_period = session.get_measurement_period()
+            assert measurement_period == 0.0071333333333333335  # 0.0071333333333333335 is the time required for 4072 to take a DC_VOLT measurement with range 10V on Digits_resolution 5.5
```

Comment:
Checks for equality with floating point numbers are brittle. See this [StackOverflow](https://stackoverflow.com/questions/5595425/what-is-the-best-way-to-compare-floats-for-almost-equality-in-python) link for a possible solution.